packages/open-payments/src/client/incoming-payment.ts

```diff
@@ -17,14 +17,80 @@ export const createIncomingPaymentRoutes = (
 ): IncomingPaymentRoutes => {
   const { axiosInstance, openApi, logger } = clientDeps
 
-  const getIncomingPaymentValidator =
+  const getIncomingPaymentOpenApiValidator =
     openApi.createResponseValidator<IncomingPayment>({
       path: getPath('/incoming-payments/{id}'),
       method: HttpMethod.GET
     })
 
   return {
     get: (args: GetArgs) =>
-      get({ axiosInstance, logger }, args, getIncomingPaymentValidator)
+      getIncomingPayment(
+        { axiosInstance, logger },
+        args,
+        getIncomingPaymentOpenApiValidator
+      )
   }
 }
+
+export const getIncomingPayment = async (
+  clientDeps: Pick<ClientDeps, 'axiosInstance' | 'logger'>,
+  args: GetArgs,
+  validateOpenApiResponse: ResponseValidator<IncomingPayment>
+) => {
+  const { axiosInstance, logger } = clientDeps
+  const { url } = args
+
+  const incomingPayment = await get(
+    { axiosInstance, logger },
+    args,
+    validateOpenApiResponse
+  )
+
+  try {
+    return validateIncomingPayment(incomingPayment)
+  } catch (error) {
+    const errorMessage = 'Could not validate incoming payment'
+    logger.error({ url, validateError: error?.message }, errorMessage)
+
+    throw new Error(errorMessage)
+  }
+}
+
+export const validateIncomingPayment = (
+  payment: IncomingPayment
+): IncomingPayment => {
+  if (payment.incomingAmount) {
+    const { incomingAmount, receivedAmount } = payment
+    if (
+      incomingAmount.assetCode !== receivedAmount.assetCode ||
+      incomingAmount.assetScale !== receivedAmount.assetScale
+    ) {
+      throw new Error(
+        'Incoming amount asset code or asset scale does not match up received amount'
+      )
+    }
+    if (BigInt(incomingAmount.value) < BigInt(receivedAmount.value)) {
+      throw new Error('Received amount is larger than incoming amount')
+    }
+    if (incomingAmount.value === receivedAmount.value && !payment.completed) {
+      throw new Error(
+        'Incoming amount matches received amount but payment is not completed'
+      )
+    }
+  }
+  if (typeof payment.ilpStreamConnection === 'object') {
```

Comment:
```suggestion
```
This check isn't necessary for now, since we're currently only dealing with:
https://github.com/interledger/rafiki/blob/2c40d222006e5beef2fe850d0208c27c70bfcc69/packages/open-payments/src/types.ts#L6-L7